utils/useInfiniteProps.ts

```diff
@@ -0,0 +1,15 @@
+import { useCallback } from "react";
+
+function useInfiniteProps<Data>(
+  data: Data[][] | undefined,
+  size: number,
+  setSize: (size: number | ((size: number) => number)) => Promise<unknown>
+) {
+  const [prev] = data?.slice(-1) ?? [];
+  const hasNextMore = prev === undefined || prev.length > 0;
+  const loading = data?.length !== size;
+  const onLoadMore = useCallback(() => setSize((n) => n + 1), [setSize]);
+  return { loading, hasNextMore, onLoadMore };
```

Comment:
私のローカル環境で最初のperPage分しか取得されなかったので眺めていたのですが、ここの変数名が古いままでしょうか

```suggestion
  const hasNextPage = prev === undefined || prev.length > 0;
  const loading = data?.length !== size;
  const onLoadMore = useCallback(() => setSize((n) => n + 1), [setSize]);
  return { loading, hasNextPage, onLoadMore };
```

Reply:
そうですね! hasNextMoreなんてものは存在しません。
感謝!